src/e-pr.js

```diff
@@ -188,5 +195,231 @@ program
+        if (error.code !== 'ENOENT') {
+          throw error;
+        }
+      }
+
+      // Create the directory
+      await fs.promises.mkdir(defaultDir, { recursive: true });
+
+      outputDir = defaultDir;
+    }
+
+    console.log(
+      `Downloading artifact '${artifactName}' from pull request #${pullRequestNumber}...`,
+    );
+
+    // Download the artifact to a temporary directory
+    const tempDir = path.join(os.tmpdir(), 'electron-tmp');
+    await fs.promises.mkdir(tempDir);
+
+    const { url } = await octokit.actions.downloadArtifact.endpoint({
+      owner: 'electron',
+      repo: 'electron',
+      artifact_id: artifact.id,
+      archive_format: 'zip',
+    });
+
+    const response = await fetch(url, {
+      headers: {
+        Authorization: `Bearer ${auth}`,
+      },
+    });
+
+    if (!response.ok) {
+      fatal(`Could not find artifact: ${url} got ${response.status}`);
+    }
+
+    const total = parseInt(response.headers.get('content-length'), 10);
+    const artifactDownloadStream = Readable.fromWeb(response.body);
+
+    try {
+      const artifactZipPath = path.join(tempDir, `${artifactName}.zip`);
+      const artifactFileStream = fs.createWriteStream(artifactZipPath);
+      await pipeline(
+        artifactDownloadStream,
+        // Show download progress
+        ...(process.env.CI ? [] : [progressStream(total, '[:bar] :mbRateMB/s :percent :etas')]),
+        artifactFileStream,
+      );
+
+      // Extract artifact zip
+      d('unzipping artifact to %s', tempDir);
+      await extractZip(artifactZipPath, { dir: tempDir });
+
+      // Check if dist.zip exists within the extracted artifact
+      const distZipPath = path.join(tempDir, 'dist.zip');
+      if (!(await fs.promises.stat(distZipPath).catch(() => false))) {
+        throw new Error(`dist.zip not found in build artifact.`);
+      }
+
+      // Extract dist.zip
+      // NOTE: 'extract-zip' is used as it correctly extracts symlinks.
+      d('unzipping dist.zip to %s', outputDir);
+      await extractZip(distZipPath, { dir: outputDir });
+
+      const platformExecutables = {
+        win32: 'electron.exe',
+        darwin: 'Electron.app/',
+        linux: 'electron',
+      };
+
+      const executableName = platformExecutables[options.platform];
+      if (!executableName) {
+        throw new Error(`Unable to find executable for platform '${options.platform}'`);
+      }
+
+      const executablePath = path.join(outputDir, executableName);
+      if (!(await fs.promises.stat(executablePath).catch(() => false))) {
+        throw new Error(`${executableName} not found within dist.zip.`);
+      }
+
+      // Cleanup temporary files
+      await fs.promises.rm(tempDir, { recursive: true });
+
+      console.log(`${color.success} Downloaded to ${outputDir}`);
+    } catch (error) {
+      // Cleanup temporary files
+      try {
+        await fs.promises.rm(tempDir, { recursive: true });
+      } catch {
+        // ignore
+      }
```

Comment:
I think this can just go in a `finally` block rather than repeating it in both blocks?

Reply:
The issue is that `fatal()` will call process.exit(1)` and the cleanup won't happen. I'll refactor this to use `process.exitCode = 1`